server/plugin/api.go

```diff
@@ -1187,6 +1193,66 @@ func (p *Plugin) handleSubscriptionNotifications(w http.ResponseWriter, r *http.
 	returnStatusOK(w)
 }
 
+func (p *Plugin) handlePipelineCommentModal(w http.ResponseWriter, r *http.Request) {
+	mattermostUserID := r.Header.Get(constants.HeaderMattermostUserID)
+	decoder := json.NewDecoder(r.Body)
+	postActionIntegrationRequest := &model.PostActionIntegrationRequest{}
+	if err := decoder.Decode(&postActionIntegrationRequest); err != nil {
+		// TODO: prevent posting any error messages except oAuth in DM for now and use dialog for all such cases
+		p.handlePipelineApprovalRequestUpdateError("Error decoding PostActionIntegrationRequest param: ", mattermostUserID, err)
+		p.handleError(w, r, &serializers.Error{Code: http.StatusInternalServerError, Message: err.Error()})
```

Comment:
this should internal server error or a bad request (400 series) as we are not able to decode ?

Reply:
This API is called by us internally. I guess that's why we are using `InternalServerError` here.
Can you please confirm @avas27JTG